packages/core/src/amazonq/explorer/amazonQChildrenNodes.ts

```diff
@@ -25,12 +27,21 @@ export const qExtensionPageCommand = Commands.declare('aws.toolkit.amazonq.exten
     void vscode.env.openExternal(vscode.Uri.parse(`vscode:extension/${VSCODE_EXTENSION_ID.amazonq}`))
 })
 
-export const dismissQTree = Commands.declare('aws.toolkit.amazonq.dismiss', () => async () => {
-    await globals.context.globalState.update(amazonQDismissedKey, true)
-    await vscode.commands.executeCommand('setContext', amazonQDismissedKey, true)
-})
+export const dismissQTree = Commands.declare(
+    { id: '_aws.toolkit.amazonq.dismiss', compositeKey: { 1: 'source' } },
+    () => async (source: string) => {
+        await telemetry.toolkit_invokeAction.run(async () => {
+            telemetry.record({
+                source: ExtensionUse.instance.isFirstUse() ? ExtStartUpSource.FirstStartUp : ExtStartUpSource.None,
```

Comment:
We've been avoiding using Enums (`ExtStartUpSource`) in favor of an object with `const`. There are some quirks with it, and it is better to use the basic types when possible.

Example of how we do it in our code:
```
export const AuthSources = {
    addConnectionQuickPick: 'addConnectionQuickPick',
    firstStartup: 'firstStartup',
    codecatalystDeveloperTools: 'codecatalystDeveloperTools',
    vscodeComponent: vscodeComponent,
    cwQuickPick: cwQuickPickSource,
    cwTreeNode: cwTreeNodeSource,
    amazonQChat: amazonQChatSource,
    authNode: 'authNode',
} as const

export type AuthSource = (typeof AuthSources)[keyof typeof AuthSources]
```

Isn't a blocker for this PR, but is probably worth to update at some point. Your code won't have to change much though. `ExtStartUpSource` can just become the `AuthSources` equivalent, and then make a type from that.